src/rules/requests/request-handlers.ts

```diff
@@ -465,7 +464,7 @@ export class PassThroughHandler extends PassThroughHandlerDefinition {
                 hostHeader[1] = updateHostHeader;
             } // Otherwise: falsey means don't touch it.
 
-            reqUrl = new URL(`${protocol}//${hostname}${(port ? `:${port}` : '')}${path}`).toString();
+            reqUrl = new URL(`${protocol}//${hostname}${(port ? `:${port}` : '')}${pathname}${search}`).toString();
```

Comment:
Unfortunately, there are other differences here where the `url.parse` output is preferable... One notable example is a simple `?` at the end of the path:

```js
console.log(url.parse('https://example.com/?').search);
// '?'

console.log(new URL('https://example.com/?').search)
// ''
```

These might be semantically equivalent in theory, but they're not actually the same in practice (a server will literally receive a different string, and can legitimately do different things based on that if it so chooses). With this change, when proxying a request to that URL through Node, the client will send a slightly different request to the server that it expected. For more context, there's a WHATWG discussion on this that you might find interesting at https://github.com/whatwg/url/issues/779 (I've just added a comment there with my perspective).

This is not great, but of course you do make a good argument about the other differences where `url.parse` is wrong too! I think maybe we might want to make our own URL parser that wraps the two to use WHATWG parsing as standard but with a few extra checks for details like this that get lost otherwise. What do you think?